### PR TITLE
TST: Metadata / MetadataColumn writer unit tests

### PR DESCRIPTION
**Pull request by jairideout**

### CODE REVIEW

File: qiime2/metadata/tests/test_io.py [hunk 1]
```diff
@@ -777,5 +779,332 @@ def test_column_types_override_directive(self):
...
+        )
+
+        self.assertEqual(obs, exp)
+
+    def test_various_numbers(self):
+        numbers = [
+            0.0, -0.0, np.nan, 1.0, 42.0, -33.0, 1e-10, 1.5e15, 0.0003, -4.234,
+            # This last number should be rounded because it exceeds 15 digits
+            # of precision.
+            12.34567891234567
```
Comment (thermokarst) [inline qiime2/metadata/tests/test_io.py]: 👍

File: qiime2/metadata/tests/test_io.py [hunk 2]
```diff
@@ -777,5 +779,332 @@ def test_column_types_override_directive(self):
...
+            "id\t42.0\t1000\t-4.2\n"
+            "#q2:types\tnumeric\tcategorical\tnumeric\n"
+            "0.000001\t2\tb\t2.5\n"
+            "0.004000\t1\tb\t4.2\n"
+            "0.000000\t3\tc\t-9.999\n"
+        )
+
+        self.assertEqual(obs, exp)
+
+    # A couple of basic tests for CategoricalMetadataColumn and
```
Comment (thermokarst) [inline qiime2/metadata/tests/test_io.py]: 👍

File: qiime2/metadata/io.py [hunk 3]
```diff
@@ -378,11 +378,40 @@ def write(self, filepath):
...
+            # References:
+            #
+            # - https://stackoverflow.com/a/2440786/3776794
+            # - https://stackoverflow.com/a/2440708/3776794
+            # - https://docs.python.org/3/library/string.html#
+            #       format-specification-mini-language
+            # - https://stackoverflow.com/a/20586479/3776794
+            # - https://drj11.wordpress.com/2007/07/03/python-poor-printing-
+            #       of-floating-point/
+            return '{0:.15g}'.format(value)
```
Comment (thermokarst) [inline qiime2/metadata/io.py]: 🎉 🎉 🎉 

Thanks for the research and notes on this!

**Comment by jairideout** [comment]
Thanks for reviewing the PR and testing out the float formatter @thermokarst!

**Pull request merged**